### PR TITLE
Update to CRD docs

### PR DESCRIPTION
**Pull request by lkysow**

### CODE REVIEW

**Review by DerekStrickland — APPROVED** [review]
LGTM.  One suggestion. No show stoppers.

File: website/pages/docs/k8s/crds.mdx [hunk 1]
```diff
@@ -307,14 +307,15 @@ spec:
...
-   define which service the intention applies to. It can be set to any value
-   however we recommend setting it to the name of the destination service the
-   intention is configuring.
-
+1. **Unlike the other configuration entries**, the value of `metadata.name` does not
+   define which service the intention applies to.
    Instead, the name of the service this intention
    is configuring is set in `spec.destination.name`.
 
+   `metadata.name` can be set to any value,
```
Comment (lkysow) [inline website/pages/docs/k8s/crds.mdx:315]:
```suggestion
   The `metadata.name` field can be set to any value,
```

**Comment by david-yu** [comment]
Added some changes to this branch after testing the CRD docs. Need to figure out how to inline suggest again.

**Comment by lkysow** [comment]
> Added some changes to this branch after testing the CRD docs. Need to figure out how to inline suggest again.

Thanks! I think https://github.com/hashicorp/consul/pull/8920#discussion_r504153781 is the only remaining thing unresolved? Up to you if you want to change that. The code will work with 1.8.4 and 1.8.5 (except for service-intentions and resolver lb config).

**Comment by david-yu** [comment]
@lkysow I would be more comfortable with making it all dependent on 1.9.0 since supporting previous releases are tricky and we want to incentivize folks to upgrade to newer Consul versions to make the upgrade path smoother. 

I think we can internally in the code comment that CRDs work with 1.8.4+.

**Comment by lkysow** [comment]
@david-yu discussed offline and we'll leave the version notes for now.

**Pull request merged**